### PR TITLE
DEV-491: Create a README.md file to document the Qt Launcher

### PR DESCRIPTION
**Pull request by danteruiz**
Ticket - https://highfidelity.atlassian.net/browse/DEV-2679

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

File: launchers/qt/BUILD.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Dependencies
+- [cmake](https://cmake.org/download/):  3.9
```
Comment (zfox23) [inline launchers/qt/BUILD.md]: I'm guessing that another dependency is Visual Studio 2019 for Windows, right?

File: launchers/qt/BUILD.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+# Windows
+cmake -G "Visual Studio 16 2019" ..
```
Comment (zfox23) [inline launchers/qt/BUILD.md]:
```suggestion
`cmake -G "Visual Studio 16 2019" ..`
```

File: launchers/qt/BUILD.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+# OSX
+cmake -G Xcode ..
```
Comment (zfox23) [inline launchers/qt/BUILD.md]:
```suggestion
`cmake -G Xcode ..`
```

File: launchers/qt/BUILD.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Dependencies
+- [cmake](https://cmake.org/download/):  3.9
+
+# Windows
+cmake -G "Visual Studio 16 2019" ..
+
+# OSX
```
Comment (zfox23) [inline launchers/qt/BUILD.md]:
```suggestion
# macOS
```

File: launchers/qt/BUILD.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+cmake -G Xcode ..
+
+
+if you wish to not use the compiled qml files pass the `-DLAUNCHER_SOURCE_TREE_RESOURCES=On` argument to cmake.
```
Comment (zfox23) [inline launchers/qt/BUILD.md]:
```suggestion
If you wish to not use the compiled QML files, pass the `-DLAUNCHER_SOURCE_TREE_RESOURCES=On` argument to cmake.
```

File: launchers/qt/readme.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# HQ Launcher
+Behavior of the HQ Launcher is as follows:
```
Comment (zfox23) [inline launchers/qt/readme.md]: - is this in reverse chronological order?
- what about login behavior?

File: launchers/qt/readme.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+* Launching the user in the current HQ domain
+* Update the Interface client to the current version.
+* Update the HQ Launcher to the current version
+
+# directory structure
+
+## src/ - contains the c++ and objective-c.
+* LauncherState - hold majority of the logic of the launcher (signin, config file, updating, running launcher)
```
Comment (zfox23) [inline launchers/qt/readme.md]: can you please explain a bit more what the "config file" does?

File: launchers/qt/readme.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# HQ Launcher
+Behavior of the HQ Launcher is as follows:
+* Launching the user in the current HQ domain
+* Update the Interface client to the current version.
+* Update the HQ Launcher to the current version
+
+# directory structure
+
+## src/ - contains the c++ and objective-c.
```
Comment (zfox23) [inline launchers/qt/readme.md]: - can you please put these in ABC order so it's easier for someone looking thru the directory to follow the list?
- it looks like this list is missing some files. is there a reason for that?
- can you please surround filenames or directory names in backticks so they stand out as code-like?

File: launchers/qt/readme.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+* LauncherState - hold majority of the logic of the launcher (signin, config file, updating, running launcher)
+* LauncherInstaller_windows - logic of how to install/uninstall HQ Launcher on windows
+* Helper - helper functions
+* UserSettings - getting the users setting (home location) from metaverse
+* BuildsRequest - getting / parsing the build info from thunder api
+* LoginRequest - checks the login credentials the user typed in. 
+* Unzipper - helper class for extracting zip files
+
+## resources/
+* image/ - Holds the images and icon that are used by the launcher
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
* images/ - Holds the images and icon that are used by the launcher
```

File: launchers/qt/readme.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+* qml/ - UI elements - `QML_FILE_FOR_UI_STATE` varible in LauchherState defines what qml files are used by the laucnher.
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
* `qml/`
  * UI elements
  * `QML_FILE_FOR_UI_STATE` variable in `LauncherState` defines what QML files are used by the launcher.
```

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22694/HighFidelity-Beta-PR16537-d8ba056.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22694/HighFidelity-Quest-Beta-PR16537-d8ba056.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22694/HighFidelity-Beta-PR16537-d8ba056.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22694/HighFidelity-Beta-Interface-PR16537-d8ba056.dmg

File: launchers/qt/readme.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# HQ Launcher
+Behavior of the HQ Launcher is as follows:
+* Update the HQ Launcher to the latest version
+* Sign up and sign in is the user is not
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
* Sign up or sign in if the user is not already signed in
```

File: launchers/qt/readme.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+* Download the latest Interface client
+* Launching the user in the current HQ domain
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
* Launch the user in the current HQ domain
```

File: launchers/qt/readme.md (hunk 8 above)
Comment (zfox23) [reply launchers/qt/readme.md]: new version looks fantastic, thank you.

File: launchers/qt/readme.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+* `Helper_windows` - helper function that depend on windows api
+* `Launcher` - initialized the Launcher Application and resources
+* `LauncherInstaller_windows` - logic of how to install/uninstall HQ Launcher on windows
+* `LauncherState` - hold majority of the logic of the launcher (signin, config file, updating, running launcher)
+  * config files hold the following saved data
+    * logged in
+    * home location
+* `LauncherWindows` - wrapper for `QQuickWindow` that implements drag feature
+* `LoginRequest` - checks the login credentials the user typed in.
+* `NSTask+NSTaskExecveAddtions` - Extension of NSTask for replacing Launcher process with interface client process
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
* `NSTask+NSTaskExecveAdditions` - Extension of NSTask for replacing Launcher process with interface client process
```

File: launchers/qt/readme.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+* `PathUtils` - Helper class for getting relative paths for HQ Launcher
+* `SignupRequest` - Determines if the users request to signup for a new account succeeded based on the entered credentials
+* `Unzipper` - helper class for extracting zip files
+* `UserSettingsRequest` - getting the users setting (home location) from metaverse
+
+## resources/
+* `images/`-  Holds the images and icon that are used by the launcher
+* `qml/`
+  * UI elements
+  * `QML_FILE_FOR_UI_STATE` varible in LauchherState defines what qml files are used by the laucnher.
```
Comment (zfox23) [inline launchers/qt/readme.md]:
```suggestion
  * `QML_FILE_FOR_UI_STATE` variable in `LauncherState` defines what QML files are used by the Launcher.
```

**Pull request merged**